### PR TITLE
Enhance description

### PR DESCRIPTION
**Pull request by KasiaFoszcz**
- Description of methods

### CODE REVIEW

File: metale_amorficzne/pages/02_Data.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
+from metale_amorficzne.streamlit_utils import show_markdown
+
+show_markdown(__file__, unsafe_allow_html=True)
```
Comment (MarekPikula) [inline metale_amorficzne/pages/02_Data.py]: Here you can skip `unsafe_allow_html`, as no explicit HTML seems to be included in `01_Data.md`.

File: metale_amorficzne/pages/02_Data.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+## Materials and research methods used for this study (Data)
+
+- Zirconium Copper (ZrCu) and Zirconium Titanium (ZrTi) alloys,
+- nanoindentation of specifically prepared, polished metal samples in the form
+  of thin coins. Indentation was performed on each sample in a matrix of 15×15
+  or 10×10 points in distances ranging from 5 to 20 um, force applied ranging
+  from 25mN to 200mN.
+- data output from nanoindentation machine *CSM Instruments NHT2* comes as .TXT
+  file and contains calculations of 21 parameters for each indentation point,
+---
```
Comment (MarekPikula) [inline metale_amorficzne/pages/02_Data.md]: After splitting files, I think that you don't need to use the horizontal split lines.

File: metale_amorficzne/pages/02_Data.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Methods and Results
+
+I prepared a parser for output files from the *CSM Instruments NHT2*
+nanoindentation machine. Initially I tried classification, but after a while of
+trials with decision trees, I realised that in this case clustering should be
+performed, as the data in unlabelled.
+
+---
```
Comment (MarekPikula) [inline metale_amorficzne/pages/02_Data.md]: Don't use horizontal lines at the end of the file.

File: metale_amorficzne/pages/03_Hierarchical clustering.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+## Limitations
+
+- Can be computationally expensive and slow for large datasets
+- May not work well with data that has complex shapes or overlapping clusters
+- Can be sensitive to the choice of similarity metric and linkage method
+- Produces a static dendrogram that cannot be easily updated as new data is
+  added
+
+---
```
Comment (MarekPikula) [inline metale_amorficzne/pages/03_Hierarchical clustering.md]: Remove hline.

File: metale_amorficzne/pages/03_Hierarchical clustering.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+successively merged together. The linkage criteria determines the metric used
+for the merge strategy:
+
+- Ward minimizes the sum of squared differences within all clusters. It is a
+  variance-minimizing approach and in this sense is similar to the- k-means
+  objective function but tackled with an agglomerative hierarchical approach.-
+- Maximum or complete linkage minimizes the maximum distance between
+  observations of pairs of clusters.-
+- Average linkage minimizes the average of the distances between all
+  observations of pairs of clusters.-
```
Comment (MarekPikula) [inline metale_amorficzne/pages/03_Hierarchical clustering.md]: What are those dashes at the ends of lines?

File: metale_amorficzne/pages/03_Hierarchical clustering.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+- Can handle different types of data, including numerical and categorical data
+- Can be used in a wide range of applications
+
+## Limitations
+
+- Can be computationally expensive and slow for large datasets
+- May not work well with data that has complex shapes or overlapping clusters
+- Can be sensitive to the choice of similarity metric and linkage method
+- Produces a static dendrogram that cannot be easily updated as new data is
+  added
```
Comment (MarekPikula) [inline metale_amorficzne/pages/03_Hierarchical clustering.md]: Missing period mark at the end of a sentence.

File: metale_amorficzne/pages/04_KMeans clustering.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+- Can be used in a wide range of applications
+
+## Limitations
+
+- Requires the number of clusters to be specified in advance
+- Can be sensitive to the initial placement of cluster centroids
+- May not work well with data that has complex shapes or overlapping clusters
+- Can be affected by outliers or noise in the data
+
+---
```
Comment (MarekPikula) [inline metale_amorficzne/pages/04_KMeans clustering.md]: Remove hline.

File: metale_amorficzne/pages/03_Hierarchical clustering.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
+from metale_amorficzne.streamlit_utils import show_markdown
+
+show_markdown(__file__, unsafe_allow_html=True)
```
Comment (MarekPikula) [inline metale_amorficzne/pages/03_Hierarchical clustering.py]: No need for `unsafe_allow_html`.

File: metale_amorficzne/pages/04_KMeans clustering.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+- Fast and scalable, making it suitable for large datasets
+- Can handle different types of data, including numerical and categorical data
+- Can be used in a wide range of applications
+
+## Limitations
+
+- Requires the number of clusters to be specified in advance
+- Can be sensitive to the initial placement of cluster centroids
+- May not work well with data that has complex shapes or overlapping clusters
+- Can be affected by outliers or noise in the data
```
Comment (MarekPikula) [inline metale_amorficzne/pages/04_KMeans clustering.md]: Missing period marks at the ends of sentences.

File: metale_amorficzne/pages/04_KMeans clustering.py [hunk 10]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
+from metale_amorficzne.streamlit_utils import show_markdown
+
+show_markdown(__file__, unsafe_allow_html=True)
```
Comment (MarekPikula) [inline metale_amorficzne/pages/04_KMeans clustering.py]: No need for `unsafe_allow_html`.

File: metale_amorficzne/pages/05_Other methods.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
+# Other methods
+## ChatGPT clustering
+
+ I read the article and used insights from the author to prepare a prompt for my
+ data: *How I Won Singapore’s GPT-4 Prompt Engineering Competition. A deep dive
+ into the strategies I learned for harnessing the power of Large Language Models
+ (LLMs)* https://towardsdatascience.com/how-i-won-singapores-gpt-4-prompt-engineering-competition-34c195a93d41#183c
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.md]: Use a proper Markdown link:

```suggestion
 data: [*How I Won Singapore’s GPT-4 Prompt Engineering Competition. A deep dive
 into the strategies I learned for harnessing the power of Large Language Models
 (LLMs)*](https://towardsdatascience.com/how-i-won-singapores-gpt-4-prompt-engineering-competition-34c195a93d41#183c)
```

File: metale_amorficzne/pages/05_Other methods.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
+# Other methods
+## ChatGPT clustering
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.md]:
```suggestion
# Other methods

## ChatGPT clustering
```

File: metale_amorficzne/Home.py [hunk 13]
```diff
@@ -0,0 +1,5 @@
+import streamlit as st
+from pathlib import Path
```
Comment (MarekPikula) [inline metale_amorficzne/Home.py]: Unused imports

File: metale_amorficzne/pages/02_Data.py [hunk 14]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
```
Comment (MarekPikula) [inline metale_amorficzne/pages/02_Data.py]: Unused import.

File: metale_amorficzne/pages/03_Hierarchical clustering.md [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+  variance-minimizing approach and in this sense is similar to the- k-means
+  objective function but tackled with an agglomerative hierarchical approach.-
+- Maximum or complete linkage minimizes the maximum distance between
+  observations of pairs of clusters.-
+- Average linkage minimizes the average of the distances between all
+  observations of pairs of clusters.-
+- Single linkage minimizes the distance between the closest observations of
+  pairs of clusters.
+
+
```
Comment (MarekPikula) [inline metale_amorficzne/pages/03_Hierarchical clustering.md]: Remove one newline.

File: metale_amorficzne/pages/03_Hierarchical clustering.py [hunk 16]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
```
Comment (MarekPikula) [inline metale_amorficzne/pages/03_Hierarchical clustering.py]: Unused import.

File: metale_amorficzne/pages/04_KMeans clustering.py [hunk 17]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
```
Comment (MarekPikula) [inline metale_amorficzne/pages/04_KMeans clustering.py]: Unused import.

File: metale_amorficzne/pages/05_Other methods.md [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+
+ I read the article and used insights from the author to prepare a prompt for my
+ data: *How I Won Singapore’s GPT-4 Prompt Engineering Competition. A deep dive
+ into the strategies I learned for harnessing the power of Large Language Models
+ (LLMs)* https://towardsdatascience.com/how-i-won-singapores-gpt-4-prompt-engineering-competition-34c195a93d41#183c
+ I limited the data input to four chosen parameters for one metal sample. The
+ result was not completely wrong, not precise though to continue trials with
+ this method.
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.md]: The entire block is indented by one space. Remove the indentation.

File: metale_amorficzne/pages/05_Other methods.md [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+ <Annex> Give a table of the list of all row numbers belonging to each cluster,
+ in order to back up your analysis. Use these table headers: [[CLUSTER_NAME],
+ List of Rows].
+
+ #############
+
+ # START ANALYSIS #
+ If you understand, ask me for my dataset.
+```
+
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.md]: Add a note that following block is the Chat's response.

File: metale_amorficzne/pages/05_Other methods.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+- Does not require the number of clusters to be specified in advance
+- Can be used in a wide range of applications
+
+### Limitations
+
+- Requires the specification of two parameters: epsilon and the minimum number
+  of points required to form a cluster
+- Can be sensitive to the choice of parameters and the distance metric used
+- May not work well with data that has varying densities or complex shapes
+- Can be computationally expensive for large datasets
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.md]: Periods

File: metale_amorficzne/pages/05_Other methods.md [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+
+In summary, Density-based clustering is a powerful type of clustering algorithm
+that can identify clusters based on the density of data points.
+
+---
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.md]: Remove hline.

File: metale_amorficzne/pages/05_Other methods.py [hunk 22]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.py]: Unused import.

File: metale_amorficzne/pages/05_Other methods.py [hunk 23]
```diff
@@ -0,0 +1,4 @@
...
+from metale_amorficzne.streamlit_utils import show_markdown
+
+show_markdown(__file__, unsafe_allow_html=True)
```
Comment (MarekPikula) [inline metale_amorficzne/pages/05_Other methods.py]: No need for `unsafe_allow_html`.

File: metale_amorficzne/pages/06_Discussion.md [hunk 24]
```diff
@@ -0,0 +1,2 @@
+# Discussion
+---
```
Comment (MarekPikula) [inline metale_amorficzne/pages/06_Discussion.md]:
```suggestion

TODO: Add discussion section.
```

File: metale_amorficzne/pages/06_Discussion.py [hunk 25]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
```
Comment (MarekPikula) [inline metale_amorficzne/pages/06_Discussion.py]: Unused import.

File: metale_amorficzne/pages/06_Discussion.py [hunk 26]
```diff
@@ -0,0 +1,4 @@
...
+from metale_amorficzne.streamlit_utils import show_markdown
+
+show_markdown(__file__, unsafe_allow_html=True)
```
Comment (MarekPikula) [inline metale_amorficzne/pages/06_Discussion.py]: No need for `unsafe_allow_html`.

File: metale_amorficzne/pages/07_References.py [hunk 27]
```diff
@@ -0,0 +1,4 @@
+import streamlit as st
```
Comment (MarekPikula) [inline metale_amorficzne/pages/07_References.py]: Unused import.

File: metale_amorficzne/pages/07_References.py [hunk 28]
```diff
@@ -0,0 +1,4 @@
...
+from metale_amorficzne.streamlit_utils import show_markdown
+
+show_markdown(__file__, unsafe_allow_html=True)
```
Comment (MarekPikula) [inline metale_amorficzne/pages/07_References.py]: No need for `unsafe_allow_html`.

**Review by MarekPikula — CHANGES_REQUESTED** [review]

File: metale_amorficzne/streamlit_utils.py [hunk 29]
```diff
@@ -0,0 +1,11 @@
...
+from pathlib import Path
+
+
+def show_markdown(source_name: str, unsafe_allow_html: bool = False, **kwargs):
+    source_file = Path(source_name)
+    st.markdown(
+        (source_file.with_suffix(".md")).read_text("utf8"),
+        unsafe_allow_html=unsafe_allow_html,
+        **kwargs
+    )
```
Comment (MarekPikula) [inline metale_amorficzne/streamlit_utils.py]: I'll clean it up a little.

File: metale_amorficzne/pages/02_Data.md (hunk 3 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/02_Data.md]: ok

File: metale_amorficzne/pages/02_Data.md (hunk 2 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/02_Data.md]: ok

File: metale_amorficzne/pages/03_Hierarchical clustering.md (hunk 15 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/03_Hierarchical clustering.md]: ok

File: metale_amorficzne/pages/03_Hierarchical clustering.md (hunk 6 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/03_Hierarchical clustering.md]: done

File: metale_amorficzne/pages/03_Hierarchical clustering.md (hunk 4 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/03_Hierarchical clustering.md]: ok

File: metale_amorficzne/pages/03_Hierarchical clustering.md (hunk 5 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/03_Hierarchical clustering.md]: done

File: metale_amorficzne/pages/04_KMeans clustering.md (hunk 9 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/04_KMeans clustering.md]: done

File: metale_amorficzne/pages/04_KMeans clustering.md (hunk 7 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/04_KMeans clustering.md]: done

File: metale_amorficzne/pages/05_Other methods.md (hunk 12 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.md]: done

File: metale_amorficzne/pages/05_Other methods.md (hunk 20 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.md]: done

File: metale_amorficzne/pages/05_Other methods.md (hunk 21 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.md]: done

File: metale_amorficzne/pages/05_Other methods.md (hunk 18 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.md]: done

File: metale_amorficzne/pages/06_Discussion.md (hunk 24 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/06_Discussion.md]: ok

File: metale_amorficzne/pages/05_Other methods.md (hunk 19 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.md]: ok

File: metale_amorficzne/pages/05_Other methods.md (hunk 11 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.md]: done

File: metale_amorficzne/pages/02_Data.py (hunk 1 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/02_Data.py]: ok

File: metale_amorficzne/pages/02_Data.py (hunk 14 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/02_Data.py]: ok

File: metale_amorficzne/pages/03_Hierarchical clustering.py (hunk 16 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/03_Hierarchical clustering.py]: ok

File: metale_amorficzne/pages/03_Hierarchical clustering.py (hunk 8 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/03_Hierarchical clustering.py]: ok

File: metale_amorficzne/pages/05_Other methods.py (hunk 22 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.py]: ok

File: metale_amorficzne/pages/05_Other methods.py (hunk 23 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/05_Other methods.py]: ok

File: metale_amorficzne/pages/06_Discussion.py (hunk 25 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/06_Discussion.py]: ok

File: metale_amorficzne/pages/06_Discussion.py (hunk 26 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/06_Discussion.py]: ok

File: metale_amorficzne/pages/07_References.py (hunk 27 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/07_References.py]: ok

File: metale_amorficzne/pages/07_References.py (hunk 28 above)
Comment (KasiaFoszcz) [reply metale_amorficzne/pages/07_References.py]: ok

**Review by MarekPikula — APPROVED** [review]
The text will need some additional polishing, but overall it looks good.

**Pull request merged**